src/Umbraco.Web/Editors/EntityController.cs

```diff
@@ -1,957 +1,1024 @@
+            }
+
+            //the EntityService cannot search members of a certain type, this is currently not supported and would require
+            //quite a bit of plumbing to do in the Services/Repository, we'll revert to a paged search
+
+            long total;
+            var searchResult = _treeSearcher.ExamineSearch(Umbraco, filter ?? "", type, pageSize, pageNumber - 1, out total, id);
+
+            return new PagedResult<EntityBasic>(total, pageNumber, pageSize)
+            {
+                Items = searchResult
+            };
+        }
+
+        /// <summary>
+        /// Get paged child entities by id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="type"></param>
+        /// <param name="pageNumber"></param>
+        /// <param name="pageSize"></param>
+        /// <param name="orderBy"></param>
+        /// <param name="orderDirection"></param>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public PagedResult<EntityBasic> GetPagedChildren(
+            int id,
+            UmbracoEntityTypes type,
+            int pageNumber,
+            int pageSize,
+            string orderBy = "SortOrder",
+            Direction orderDirection = Direction.Ascending,
+            string filter = "")
+        {
+            if (pageNumber <= 0)
+                throw new HttpResponseException(HttpStatusCode.NotFound);
+            if (pageSize <= 0)
+                throw new HttpResponseException(HttpStatusCode.NotFound);
+
+            var objectType = ConvertToObjectType(type);
+            if (objectType.HasValue)
+            {
+                long totalRecords;
+                var entities = Services.EntityService.GetPagedChildren(id, objectType.Value, pageNumber - 1, pageSize, out totalRecords, orderBy, orderDirection, filter);
+
+                if (totalRecords == 0)
+                {
+                    return new PagedResult<EntityBasic>(0, 0, 0);
+                }
+
+                var pagedResult = new PagedResult<EntityBasic>(totalRecords, pageNumber, pageSize)
+                {
+                    Items = entities.Select(Mapper.Map<EntityBasic>)
+                };
+
+                return pagedResult;
+            }
+
+            //now we need to convert the unknown ones
+            switch (type)
+            {
+                case UmbracoEntityTypes.PropertyType:
+                case UmbracoEntityTypes.PropertyGroup:
+                case UmbracoEntityTypes.Domain:
+                case UmbracoEntityTypes.Language:
+                case UmbracoEntityTypes.User:
+                case UmbracoEntityTypes.Macro:
+                default:
+                    throw new NotSupportedException("The " + typeof(EntityController) + " does not currently support data for the type " + type);
+            }
+        }
+
+        [Obsolete("This method is obsolete, use the overload with dataTypeId instead", false)]
```

Comment:
Same as above, this can be backwards compat by just adding `Guid? dataTypeId = null` as the last parameter